app/controllers/notifications_controller.rb

```diff
@@ -0,0 +1,91 @@
+class NotificationsController < ApplicationController
+  before_action :show_notification_count, only: :index
+
+  def index
+    if current_user.user_type == "taskee" then taskee_notifications end
+    if current_user.user_type == "tasker" then tasker_notifications end
+  end
+
+  def show
+    request = TaskManagement.find params[:id]
+    request.update_attribute(:viewed, true)
+    title = params[:title]
+    description = request.task_desc
+    date = pad_date(request.start_time.strftime("%B %e")) << " of this year"
+    time = format_for_view(request.start_time, request.end_time)
+    amount = request.amount
+
+    render json: {
+      title: title,
+      description: description,
+      date: date,
+      time: time,
+      amount: amount
+    }
+  end
+
+  def update
+    record = TaskManagement.find params[:id]
+    if record.update_attribute(:status, params[:status])
+      record.update_attribute(:tasker_notified, false)
+      notify_tasker(record.tasker_id)
+      render json: { message: "success" }
+    end
+  end
+
+  private
+
+  def notifications_for_taskee
+    @notifications ||= TaskManagement.
+                       all_notifications_for("taskee", current_user.id)
+  end
+
+  def notifications_for_tasker
```

Comment:
`notifications_for_tasker` and `notifications_for_taskee` could be combined to form a method called for example `notifications_for(user_type)`.